core/src/main/java/org/frankframework/validation/SchemaUtils.java

```diff
@@ -163,10 +164,12 @@ public static Set<IXSD> mergeXsdsGroupedByNamespaceToSchemasWithoutIncludes(ISco
 					skipRootElementStart = i != 1;
 					skipRootElementEnd = i < xsds.size();
 				}
-				xsdToXmlStreamWriter(xsd, w, false, true, skipRootElementStart, skipRootElementEnd, rootAttributes, rootNamespaceAttributes, imports, false);
+				xsdToXmlStreamWriter(xsd, w, false, true, skipRootElementStart, skipRootElementEnd, rootAttributes, rootNamespaceAttributes, imports);
 			}
 			// TODO: After creating merged XSD, while we still know what the source files are, we should now validate for duplicate elements
-			if (resultXsd != null) {
+			if (xmlStreamWriter == null) {
+				w.close();
```

Comment:
kan w niet null zijn hier?

Reply:
Deze code is nu nog eens opgeschoond en de method opgesplitst in twee methods die ieder op eigen manier de XSDs mergen, zonder alles in 1 proberen te proppen met allemaal rare switches.